File: internal/domain/tuning.go
package domain

import "time"

type TuningStudy struct {
	ID             string             `json:"id"`
	Status         string             `json:"status"`
	Error          string             `json:"error"`
	Trials         int                `json:"trials"`
	Completed      int                `json:"completed"`
	Failed         int                `json:"failed"`
	Seconds        int                `json:"seconds"`
	Elapsed        float64            `json:"elapsed"`
	BaselineRun    string             `json:"baseline_run"`
	BaselineValue  *float64           `json:"baseline_value"`
	BestRun        string             `json:"best_run"`
	BestValue      *float64           `json:"best_value"`
	BestParams     map[string]float64 `json:"best_params"`
	PendingRequest string             `json:"pending_request"`
	ManifestHash   string             `json:"manifest_hash"`
	CreatedAt      time.Time          `json:"created_at"`
	UpdatedAt      time.Time          `json:"updated_at"`
	Validations    []TuningValidation `json:"validations"`
}
type TuningTrial struct {
	StudyID    string             `json:"study_id"`
	Number     int                `json:"number"`
	RunID      string             `json:"run_id"`
	RequestID  string             `json:"request_id"`
	Params     map[string]float64 `json:"params"`
	Actual     map[string]float64 `json:"actual"`
	Status     string             `json:"status"`
	Value      *float64           `json:"value"`
	Reason     string             `json:"reason"`
	ResultHash string             `json:"result_hash"`
	Delivered  bool               `json:"delivered"`
}
type TuningValidation struct {
	BaselineRun  string `json:"baseline_run"`
	CandidateRun string `json:"candidate_run"`
	InputDir     string `json:"input_dir"`
	Error        string `json:"error"`
}

File: web/src/lib/tuning.ts
export type TuneParameter = {
  name: string; type: string; current: number; low: number | null; high: number | null;
  step: number | null; log: boolean; enabled: boolean; line: number; start: number; end: number; token: string;
}
export type TuneScan = { hash: string; source: string; parameters: TuneParameter[] }
export type TuneStudy = {
  id: string; status: string; error: string; trials: number; completed: number; failed: number;
  seconds: number; elapsed: number; baseline_run: string; baseline_value: number | null;
  best_run: string; best_value: number | null; best_params: Record<string, number>;
  validations: { baseline_run: string; candidate_run: string; input_dir: string; error: string }[];
}
export type TuneTrial = { number: number; run_id: string; status: string; params: Record<string, number>; value: number | null; reason: string }
export type TuneEnvironment = { available: boolean; ready: boolean; path: string; python: string; optuna: string; compressed_bytes: number; unpacked_bytes: number }
export type TuneDetail = { worker_log: string; study: TuneStudy; active: boolean; usage_bytes: number; manifest: { request: { solver: string; input_dir: string; threads: number; timeout_ms: number }; parameters: TuneParameter[]; prepared: { config: { File: { project: { objective: string } } }; inputs: unknown[] } } }
